packages/axios-extension/src/abap/catalog/v2-catalog-service.ts

```diff
@@ -42,15 +42,17 @@ export class V2CatalogService extends CatalogService {
      * @returns version independent information
      */
     protected mapServices(services: ODataServiceV2Info[]): ODataServiceInfo[] {
-        return services.map((service) => {
-            return {
-                id: service.ID,
-                name: service.TechnicalServiceName,
-                path: new URL(service.ServiceUrl).pathname,
-                serviceVersion: service.TechnicalServiceVersion + '',
-                odataVersion: ODataVersion.v2
-            };
-        });
+        if (Array.isArray(services)) {
```

Comment:
I am curious, what is the scenario that resulted in `mapServices` being called with something that wasn't an array of services?

Reply:
I haven't been able to test with v2 specifically but when testing `@sap-ux/axios-extension` in bas (using an ES5) system, I successfully retreived v2 servcies, but when changing the catalog request to v4 I see this error 

<img width="1179" alt="Screenshot 2022-08-19 at 11 56 20" src="https://user-images.githubusercontent.com/99677697/185604656-3cfcfbe1-939f-4621-a0d5-24b5dd1caf5c.png">

Reply:
I am able to reproduce it. Let me have a look. The required fix should not be part of this PR but separate including a unit test covering the scenario.

Reply:
Please remove your changes from the `axios-extension` and review https://github.com/SAP/open-ux-tools/pull/631. This should solve your problematic scenario.

Reply:
Merged. Please fetch the latest version of `main` and remove your change

Reply:
Updated

Reply:
THanks